File: src/Python/GeometricObjects/EllipticalCylinder.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import math

import vtk


def main():
    colors = vtk.vtkNamedColors()

    angle = 0
    r1 = 50
    r2 = 30
    centerX = 10.0
    centerY = 5.0

    points = vtk.vtkPoints()
    idx = 0
    while angle <= 2.0 * vtk.vtkMath.Pi() + (vtk.vtkMath.Pi() / 60.0):
        points.InsertNextPoint(r1 * math.cos(angle) + centerX,
                               r2 * math.sin(angle) + centerY,
                               0.0)
        angle = angle + (vtk.vtkMath.Pi() / 60.0)
        idx += 1

    line = vtk.vtkPolyLine()
    line.GetPointIds().SetNumberOfIds(idx)
    for i in range(0, idx):
        line.GetPointIds().SetId(i, i)

    lines = vtk.vtkCellArray()
    lines.InsertNextCell(line)

    polyData = vtk.vtkPolyData()
    polyData.SetPoints(points)
    polyData.SetLines(lines)

    extrude = vtk.vtkLinearExtrusionFilter()
    extrude.SetInputData(polyData)
    extrude.SetExtrusionTypeToNormalExtrusion()
    extrude.SetVector(0, 0, 100.0)
    extrude.Update()

    lineMapper = vtk.vtkPolyDataMapper()
    lineMapper.SetInputData(polyData)

    lineActor = vtk.vtkActor()
    lineActor.SetMapper(lineMapper)
    lineActor.GetProperty().SetColor(colors.GetColor3d("Peacock"))

    mapper = vtk.vtkPolyDataMapper()
    mapper.SetInputConnection(extrude.GetOutputPort())

    back = vtk.vtkProperty()
    back.SetColor(colors.GetColor3d("Tomato"))

    actor = vtk.vtkActor()
    actor.SetMapper(mapper)
    actor.GetProperty().SetColor(colors.GetColor3d("Banana"))
    actor.SetBackfaceProperty(back)

    ren = vtk.vtkRenderer()
    ren.SetBackground(colors.GetColor3d("SlateGray"))
    ren.AddActor(actor)
    ren.AddActor(lineActor)

    renWin = vtk.vtkRenderWindow()
    renWin.SetWindowName("Elliptical Cylinder")
    renWin.AddRenderer(ren)
    renWin.SetSize(600, 600)

    iren = vtk.vtkRenderWindowInteractor()
    iren.SetRenderWindow(renWin)

    style = vtk.vtkInteractorStyleTrackballCamera()
    iren.SetInteractorStyle(style)

    camera = vtk.vtkCamera()
    camera.SetPosition(0, 1, 0)
    camera.SetFocalPoint(0, 0, 0)
    camera.SetViewUp(0, 0, 1)
    camera.Azimuth(30)
    camera.Elevation(30)

    ren.SetActiveCamera(camera)
    ren.ResetCamera()
    ren.ResetCameraClippingRange()

    renWin.Render()
    iren.Start()


if __name__ == '__main__':
    main()

File: src/Java/GeometricObjects/EllipticalCylinder.java

import vtk.vtkActor;
import vtk.vtkNamedColors;
import vtk.vtkPoints;
import vtk.vtkPolyLine;
import vtk.vtkCellArray;
import vtk.vtkLinearExtrusionFilter;
import vtk.vtkPolyDataMapper;
import vtk.vtkProperty;
import vtk.vtkCamera;
import vtk.vtkNativeLibrary;
import vtk.vtkPolyData;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;



public class EllipticalCylinder  
{
	  //private static final long serialVersionUID = 1L;
	 

	  // -----------------------------------------------------------------
	  // Load VTK library and print which library was not properly loaded
	  static 
	  {
	    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
	    {
	      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
	      {
	        if (!lib.IsLoaded()) 
	        {
	          System.out.println(lib.GetLibraryName() + " not loaded");
	        }
	      }
	    }
	    vtkNativeLibrary.DisableOutputWindow(null);
	  }
	  // -----------------------------------------------------------------
	  
	  
	 
 
	  public static void main(String s[]) 
	  {
		  vtkNamedColors Color = new vtkNamedColors();
		  
		    //For Actor Color
		    double ActorColor[] = new double[4];
		    //For Line Actor Color
		    double LineActorColor[] = new double[4];
		    //For Back Color
		    double BackColor[] = new double[4];
		    //Renderer Background Color
		    double BgColor[] = new double[4];
		  
			   
		    //Change Color Name to Use your own Color for Change Actor Color
		    Color.GetColor("Cyan",ActorColor);
		    //Change Color Name to Use your own Color for Change line Actor Color
		    Color.GetColor("OrangeRed",LineActorColor);
		    //Change Color Name to Use your own Color for Back
		    Color.GetColor("OrangeRed",BackColor);
		    //Change Color Name to Use your own Color for Renderer Background
		    Color.GetColor("DimGray",BgColor);
		    
		    
		    double angle = 0;
		    double r1, r2;
		    double centerX, centerY;
		    r1 = 50;
		    r2 = 30;
		    centerX = 10.0;
		    centerY = 5.0;
		    

		    vtkPoints Points = new vtkPoints();
		    int id = 0;
		    while (angle <= 2.0 * Math.PI + (Math.PI / 60.0))
		    {
		      Points.InsertNextPoint(r1 * Math.cos(angle)+centerX,r2 * Math.sin(angle)+ centerY, 0.0);
		      angle = angle + (Math.PI / 60.0);
		      ++id;
		    }
		    
		    vtkPolyLine Line = new vtkPolyLine();
		    Line.GetPointIds().SetNumberOfIds(id);
		    for(int i = 0; i <id; ++i)
	    	  {
	    	    Line.GetPointIds().SetId(i, i);
	    	  }
		    vtkCellArray Lines = new vtkCellArray();
		    Lines.InsertNextCell(Line);
		    
		    vtkPolyData PolyData = new vtkPolyData();
		    PolyData.SetPoints(Points);
		    PolyData.SetLines(Lines);
		    
		    vtkLinearExtrusionFilter Extrude = new vtkLinearExtrusionFilter();
		    Extrude.SetInputData(PolyData);
		    Extrude.SetExtrusionTypeToNormalExtrusion();
		    Extrude.SetVector(0, 0, 100);
		    Extrude.Update();
		    
		    vtkPolyDataMapper LineMapper = new vtkPolyDataMapper();
		    LineMapper.SetInputData(PolyData);
		    
		    vtkActor LineActor = new vtkActor();
		    LineActor.SetMapper(LineMapper);
		    LineActor.GetProperty().SetColor(LineActorColor);

		    vtkPolyDataMapper Mapper = new vtkPolyDataMapper();
		    Mapper.SetInputConnection(Extrude.GetOutputPort());
		    
		    
		    vtkProperty Back = new vtkProperty();
		    Back.SetColor(BackColor);
		    
		    vtkActor Actor = new vtkActor();
		    Actor.SetMapper(Mapper);
		    Actor.GetProperty().SetColor(ActorColor);
		    Actor.SetBackfaceProperty(Back);
		    
		    vtkCamera Camera = new vtkCamera();
		    Camera.SetPosition(0, 1, 0);
		    Camera.SetFocalPoint(0, 0, 0);
		    Camera.SetViewUp(0, 0, 1);
		    Camera.Azimuth(30);
		    Camera.Elevation(30);
		    
		    
		    // Create the renderer, render window and interactor.
	        vtkRenderer ren = new vtkRenderer();
	        vtkRenderWindow renWin = new vtkRenderWindow();
	        renWin.AddRenderer(ren);
	        vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
	        iren.SetRenderWindow(renWin);
	        
	        // Visualise the arrow
	        ren.AddActor(Actor);
	        ren.AddActor(LineActor);
	        ren.SetBackground(BgColor);
	        ren.SetActiveCamera(Camera);
	        ren.ResetCamera();
	        ren.ResetCameraClippingRange();

	        renWin.SetSize(300, 300);
	        renWin.Render();
	    

	        iren.Initialize();
	        iren.Start();

	  }
}
